server/athenian/api/controllers/miners/github/contributors.py

```diff
@@ -21,94 +23,109 @@ async def mine_contributors(repos: Collection[str],
                             time_to: Optional[datetime],
                             db: databases.Database,
                             cache: Optional[aiomcache.Client],
-                            with_stats: Optional[bool] = True) -> List[dict]:
+                            with_stats: Optional[bool] = True,
+                            as_roles: List[str] = None) -> List[dict]:
     """Discover developers who made any important action in the given repositories and \
     in the given time frame."""
     time_from = time_from or datetime(1970, 1, 1, tzinfo=timezone.utc)
     now = datetime.now(timezone.utc) + timedelta(days=1)
     time_to = time_to or datetime(now.year, now.month, now.day, tzinfo=timezone.utc)
+    as_roles = as_roles or []
 
-    return await _mine_contributors(repos, time_from, time_to, with_stats, db, cache)
+    return await _mine_contributors(repos, time_from, time_to, with_stats, as_roles, db, cache)
 
 
 @sentry_span
 @cached(
     exptime=5 * 60,
     serialize=marshal.dumps,
     deserialize=marshal.loads,
-    key=lambda repos, time_from, time_to, with_stats, **_: (
-        ",".join(repos), time_from.timestamp(), time_to.timestamp(), with_stats),
+    key=lambda repos, time_from, time_to, with_stats, as_roles, **_: (
+        ",".join(repos), time_from.timestamp(), time_to.timestamp(), with_stats,
+        sorted(as_roles)),
 )
 async def _mine_contributors(repos: Collection[str],
                              time_from: datetime,
                              time_to: datetime,
                              with_stats: bool,
+                             as_roles: List[str],
                              db: databases.Database,
                              cache: Optional[aiomcache.Client]) -> List[dict]:
     assert isinstance(time_from, datetime)
     assert isinstance(time_to, datetime)
 
+    common_prs_where = and_(
+        PullRequest.repository_full_name.in_(repos),
+        PullRequest.hidden.is_(False),
+        or_(PullRequest.created_at.between(time_from, time_to),
+            and_(PullRequest.created_at < time_to,
+                 PullRequest.closed_at.is_(None)),
+            PullRequest.closed_at.between(time_from, time_to),
+            PullRequest.updated_at.between(time_from, time_to)),
```

Comment:
Is this a new product requirement? It totally makes sense to me, indeed when I was debugging this with the webapp there were some contributors being returned I didn't understand why all their PRs where closed before the `time_from`, then I realized it was because some PRs have been updated in the time interval (a label added).

\cc @warenlg

Reply:
This is how the rest of the miners work atm. I used to have `updated_at` everywhere in the past, and it was a source of inconsistencies. Just forgot to remove it here.